x/dualstaking/keeper/grpc_query_delegator_rewards.go

```diff
@@ -17,21 +17,13 @@ func (k Keeper) DelegatorRewards(goCtx context.Context, req *types.QueryDelegato
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
 	var rewards []types.DelegatorRewardInfo
-	resProviders, err := k.DelegatorProviders(goCtx, &types.QueryDelegatorProvidersRequest{Delegator: req.Delegator})
-	if err != nil {
-		return nil, err
-	}
-
-	for _, delegation := range resProviders.Delegations {
-		if delegation.Provider == req.Provider || req.Provider == "" {
-			delegatorReward, found := k.GetDelegatorReward(ctx, delegation.Provider, delegation.Delegator)
-			if found {
-				reward := types.DelegatorRewardInfo{
-					Provider: delegation.Provider,
-					Amount:   delegatorReward.Amount,
-				}
-				rewards = append(rewards, reward)
-			}
+	allRewards := k.GetAllDelegatorReward(ctx)
+	for _, delegatorReward := range allRewards {
+		if delegatorReward.Delegator == req.Delegator && (delegatorReward.Provider == req.Provider || req.Provider == "") {
```

Comment:
let's optimize the easy case: when the provider is specified call GetDelegatorReward directly instead of getting all of them and iterating through them

Reply:
done. see https://github.com/lavanet/lava/commit/384af84d950a2208dacad83ffe2dfea836a645d7